astroquery/esa/xmm_newton/core.py

```diff
@@ -266,5 +268,173 @@ def get_columns(self, table_name, *, only_names=True, verbose=False):
+        are described in details in the Pipeline Products Description
+        [XMM-SOC-GEN-ICD-0024](https://xmm-tools.cosmos.esa.int/external/xmm_obs_info/odf/data/docs/XMM-SOC-GEN-ICD-0024.pdf).
+        """
+
+        _product_type = ["IMAGE_"]
+        _instrument = ["M1", "M2", "PN", "EP"]
+        _band = [1, 2, 3, 4, 5, 8]
+        _path = ""
+        if get_detmask:
+            _product_type.append("DETMSK")
+        if get_exposure_map:
+            _product_type.append("EXPMAP")
+        if path != "" and os.path.exists(path):
+            _path = path
+
+        ret = {}
+        if band == []:
+            band = _band
+        else:
+            for i in band:
+                if i not in _band:
+                    log.warning("Invalid band %u" % i)
+                    band.remove(i)
+
+        if instrument == []:
+            instrument = _instrument
+        else:
+            for i in instrument:
+                if i not in _instrument:
+                    log.warning("Invalid instrument %s" % i)
+                    instrument.remove(i)
+        try:
+            with tarfile.open(filename, "r") as tar:
+                for i in tar.getmembers():
+                    paths = os.path.split(i.name)
+                    fname = paths[1]
+                    paths = os.path.split(paths[0])
+                    if paths[1] != "pps":
+                        continue
+                    fname_info = self._parse_filename(fname)
+                    if fname_info["X"] != "P":
+                        continue
+                    if not fname_info["I"] in instrument:
+                        continue
+                    if not int(fname_info["S"]) in band:
+                        continue
+                    if not fname_info["T"] in _product_type:
+                        continue
+                    tar.extract(i, _path)
+                    if not ret.get(int(fname_info["S"])):
+                        ret[int(fname_info["S"])] = {}
+                    b = int(fname_info["S"])
+                    ins = fname_info["I"]
+                    value = os.path.abspath(os.path.join(_path, i.name))
+                    if fname_info["T"] == "DETMSK":
+                        ins = fname_info["I"] + "_det"
+                    elif fname_info["T"] == "EXPMAP":
+                        ins = fname_info["I"] + "_expo"
+                    if ret[b].get(ins) and type(ret[b].get(ins)) == str:
+                        log.warning("More than one file found with the "
+                                    "band %u and "
+                                    "the instrument: %s" % (b, ins))
+                        ret[b][ins] = [ret[b][ins], value]
+                    elif ret[b].get(ins) and type(ret[b].get(ins)) == list:
+                        ret[b][ins].append(value)
+                    else:
+                        ret[b][ins] = value
+        except FileNotFoundError as err:
+            raise err
```

Comment:
When I said to raise an error, I meant just skip the whole try/except; it's not needed.
```suggestion
        with tarfile.open(filename, "r") as tar:
            for i in tar.getmembers():
                paths = os.path.split(i.name)
                fname = paths[1]
                paths = os.path.split(paths[0])
                if paths[1] != "pps":
                    continue
                fname_info = self._parse_filename(fname)
                if fname_info["X"] != "P":
                    continue
                if not fname_info["I"] in instrument:
                    continue
                if not int(fname_info["S"]) in band:
                    continue
                if not fname_info["T"] in _product_type:
                    continue
                tar.extract(i, _path)
                if not ret.get(int(fname_info["S"])):
                    ret[int(fname_info["S"])] = {}
                b = int(fname_info["S"])
                ins = fname_info["I"]
                value = os.path.abspath(os.path.join(_path, i.name))
                if fname_info["T"] == "DETMSK":
                    ins = fname_info["I"] + "_det"
                elif fname_info["T"] == "EXPMAP":
                    ins = fname_info["I"] + "_expo"
                if ret[b].get(ins) and type(ret[b].get(ins)) == str:
                    log.warning("More than one file found with the "
                                "band %u and "
                                "the instrument: %s" % (b, ins))
                    ret[b][ins] = [ret[b][ins], value]
                elif ret[b].get(ins) and type(ret[b].get(ins)) == list:
                    ret[b][ins].append(value)
                else:
                    ret[b][ins] = value
```

Reply:
Good morning,

Quick note. Regarding the moving of this feature to an utils.py, we have prepared two similar features that extract files from a given tar file and it would make sense to also put them there. However, one of them reads content of specific files and does queries to an FTP server to download extra data. So, would you still advise this feature to be put in this module?

Thank you
Luis